src/conventions/h-group/clue-interpretation/interpret-clue.js

```diff
@@ -79,9 +79,30 @@ function resolve_clue(game, old_game, action, inf_possibilities, focused_card) {
 	const old_inferred = old_game.common.thoughts[focused_card.order].inferred;
 
 	focus_thoughts.inferred = focus_thoughts.inferred.intersect(inf_possibilities);
+	focus_thoughts.self_connection = false;
 
-	for (const { connections, suitIndex, rank, save, interp } of inf_possibilities) {
+	for (const { connections, suitIndex, rank, save, interp, self_connection } of inf_possibilities) {
 		const inference = { suitIndex, rank };
+		focus_thoughts.self_connection ||= (self_connection?.length > 0);
+
+		// A finesse is considered important if it could only have been given by this player.
+		if (connections.some(connection => connection.type == 'finesse')) {
+			for (let i = (giver + 1) % state.numPlayers; i != target; i = (i + 1) % state.numPlayers) {
+				if (i == target)
```

Comment:
This condition will only be true if there are only 2 players in the game, right? I would prefer if this was more explicit by putting `&& state.numPlayers > 2` in the outer `if` condition instead.

Reply:
Actually, I don't think this will ever be true (the loop just won't run), but I think we can always assume that finesses in 2-player games are important.

Reply:
Apologies, you're right. This needs to loop through all players. We of course skip the target because they can't clue themselves, but a player after the target could still clue the target. Fixed.